app/assets/javascripts/components/GlobalNav.js

```diff
@@ -32,13 +32,177 @@ define(['jquery', 'DoughBaseComponent', 'mediaQueries', 'utilities'], function($
+          }
+          break;
+        // enter
+        case 13:
+          if (level == 'top') {
+            e.preventDefault();
+            self._openDesktopSubNav(e.target, false);
+            self._moveFocusToSubNav(e.target);
+          }
+          break;
+        // escape
+        case 27:
+          if (level == 'subnav') {
+            self._closeDesktopSubNav($(e.target).parents('[data-dough-subnav]'), false);
+            self._moveTopLevelFocus(e.target);
+          }
+          break;
+        // spacebar
+        case 32:
+          e.preventDefault();
+
+          if (level == 'top') {
+            self._openDesktopSubNav(e.target, false);
+          }
+          break;
+        // left arrow
+        case 37:
+          self._moveTopLevelFocus(e.target, 'left');
+
+          if (level == 'subnav') {
+            self._closeDesktopSubNav($(e.target).parents('[data-dough-subnav]'), false);
+          }
+          break;
+        // up arrow
+        case 38:
+          if (level == 'subnav') {
+            self._moveSubNavFocus(e.target, 'up');
+          }
+          break;
+        // right arrow
+        case 39:
+          self._moveTopLevelFocus(e.target, 'right');
+
+          if (level == 'subnav') {
+            self._closeDesktopSubNav($(e.target).parents('[data-dough-subnav]'), false);
+          }
+          break;
+        // down arrow
+        case 40:
+          if (level == 'top') {
+            if ($(e.target).parents('[data-dough-nav-clump]').hasClass('is-active')) {
+              self._moveFocusToSubNav(e.target);
+            } else {
+              self._openDesktopSubNav(e.target, false);
+            }
+          } else {
+            self._moveSubNavFocus(e.target, 'down');
+          }
+          break;
+      }
+    })
+  }
+
+  /**
+  * Moves focus to first link of subnav
+  */
+  GlobalNav.prototype._moveFocusToSubNav = function(el) {
+    var firstLink = $(el).siblings('[data-dough-subnav]').find('[data-dough-subcategories]').find('a')[0];
+
+    $(firstLink).focus();
+  }
+
+  /**
+  * Moves focus of subnav up or down
+  */
+  GlobalNav.prototype._moveSubNavFocus = function(el, dir) {
+    var subCatArray = $(el).parents('[data-dough-subcategories]').find('a');
+    var index = $(subCatArray).index(el);
+
+    if (dir == 'up') {
+      if (index > 0) {
+        subCatArray[index - 1].focus();
+      } else {
+        subCatArray[subCatArray.length - 1].focus();
+      }
+    } else {
+      if (index < subCatArray.length - 1) {
+        subCatArray[index + 1].focus();
+      } else {
+        subCatArray[0].focus();
+      }
+    }
+  }
+
+  /**
+  * Moves focus of top level nav to left or right
+  */
+  GlobalNav.prototype._moveTopLevelFocus = function(el, dir) {
+    var thisClump = $(el).parents('[data-dough-nav-clump]');
+    var clumps = $(thisClump).siblings('[data-dough-nav-clump]');
+    var firstClump = $(clumps).get()[0];
+    var lastClump = $(clumps).get()[clumps.length - 1];
```

Comment:
I'm not sure that `.get()` is right here. I think this is a [method that's only supported in IE11+ ](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map/get) that's intended to get items from a key/value map. So whilst it works, it's not going to give us our browser support.

Since we're working with jQuery selectors, I think it's better to use  the jQuery methods`.first()` or `.eq()` to filter the set of clumps - https://api.jquery.com/category/traversing/